nifi-nar-bundles/nifi-html-bundle/nifi-html-processors/src/main/java/org/apache/nifi/GetHTMLElement.java

```diff
@@ -0,0 +1,243 @@
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .allowableValues(ELEMENT_HTML, ELEMENT_TEXT, ELEMENT_ATTRIBUTE, ELEMENT_DATA)
+            .defaultValue(ELEMENT_HTML)
+            .build();
+
+    public static final PropertyDescriptor DESTINATION = new PropertyDescriptor.Builder()
+            .name("Destination")
+            .description("Control if element extracted is written as a flowfile attribute or " +
+                    "as flowfile content.")
+            .required(true)
+            .allowableValues(DESTINATION_ATTRIBUTE, DESTINATION_CONTENT)
+            .defaultValue(DESTINATION_ATTRIBUTE)
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(URL);
+        descriptors.add(CSS_SELECTOR);
+        descriptors.add(HTML_CHARSET);
+        descriptors.add(OUTPUT_TYPE);
+        descriptors.add(DESTINATION);
+        descriptors.add(PREPEND_ELEMENT_VALUE);
+        descriptors.add(APPEND_ELEMENT_VALUE);
+        descriptors.add(ATTRIBUTE_KEY);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_ORIGINAL);
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_NOT_FOUND);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        try {
+
+            final Document doc = parseHTMLDocumentFromFlowfile(flowFile, context, session);
+            final Elements eles = doc.select(context.getProperty(CSS_SELECTOR)
+                    .evaluateAttributeExpressions().getValue());
+            final String prependValue = context.getProperty(PREPEND_ELEMENT_VALUE)
+                    .evaluateAttributeExpressions(flowFile).getValue();
+            final String appendValue = context.getProperty(APPEND_ELEMENT_VALUE)
+                    .evaluateAttributeExpressions(flowFile).getValue();
+
+            if (eles == null || eles.size() == 0) {
+                //No element found
+                session.transfer(flowFile, REL_NOT_FOUND);
+            } else {
+                for (final Element ele : eles) {
+                    final FlowFile ff = session.create();
```

Comment:
Should be calling session.create(flowFile) so that the Provenance Lineage is handled appropriately and original FlowFile's attributes are transferred along.